### PR TITLE
feat: Add VTEX RequestCapture analytics

### PR DESCRIPTION
**Pull request by igorbrasileiro**
## What's the purpose of this pull request?
This pull request adds the Request Capture engine to the analytics backend.

## How it works? 
<em>Tell us the role of the new feature, or component, in its context.</em>

## How to test it?
<em>Describe the steps with bullet points. Is there any external reference, link, or example?</em> 

## References
<em>Spread the knowledge: is this any content you used to create this PR that is worth sharing?</em>  

<em>Extra tip: add references to related issues or mention people important to this PR may be good for the documentation and reviewing process</em>

### CODE REVIEW

File: src/sdk/analytics/utils/navigation.ts [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+function setVtexCookie(
+  cookieId: string,
+  cookieValue: string,
+  expirationTime: number
+) {
+  const formattedValue = encodeURIComponent(cookieValue)
+  const host = window.location.hostname
+  const expirationDate = new Date()
+
+  expirationDate.setTime(expirationDate.getTime() + expirationTime)
```
Comment (icazevedo) [inline src/sdk/analytics/utils/navigation.ts:38]: I think this is more readable

```suggestion
  expirationDate.setTime(new Date().getTime() + expirationTime)
```

Comment (igorbrasileiro) [reply src/sdk/analytics/utils/navigation.ts]: Fixed

File: src/sdk/analytics/utils/navigation.ts [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+ *
+ */
+function uuidv4() {
+  const uuid = 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, (c) => {
+    const r = (Math.random() * 16) | 0
+
+    return (c === 'x' ? r : (r & 0x3) | 0x8).toString(16)
+  })
+
+  return uuid
```
Comment (lariciamota) [inline src/sdk/analytics/utils/navigation.ts:19]: To stop SonarQube warning:
```suggestion
  return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, (c) => {
    const r = (Math.random() * 16) | 0

    return (c === 'x' ? r : (r & 0x3) | 0x8).toString(16)
  })
```

**Review by lariciamota — CHANGES_REQUESTED** [review]

File: src/sdk/analytics/utils/navigation.ts [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+ * Get title of the page.
+ *
+ * @returns {String} Page title.
+ */
+function getPageTitle(): string {
+  return document.title
+}
+
+/**
+ * Get account name from the environment varibable GATSBY_STORE_ID.
```
Comment (lariciamota) [inline src/sdk/analytics/utils/navigation.ts:144]:
```suggestion
 * Get account name from the environment variable GATSBY_STORE_ID.
```

**Review by lariciamota — CHANGES_REQUESTED** [review]

**Pull request closed without merging**